contrib/src/main/java/com/datatorrent/contrib/parser/StreamingJsonParser.java

```diff
@@ -0,0 +1,451 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.StringTokenizer;
+
+import org.elasticsearch.common.primitives.Ints;
+import org.json.simple.parser.JSONParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.hadoop.classification.InterfaceStability;
+
+import com.google.common.collect.Lists;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.lib.parser.Parser;
+import com.datatorrent.lib.util.FieldInfo;
+import com.datatorrent.lib.util.FieldInfo.SupportType;
+import com.datatorrent.lib.util.KeyValPair;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * Operator that parses a JSON string tuple and emits a POJO on the output port
+ * and tuples that could not be parsed on error port.Upstream operator needs to
+ * ensure that a full JSON record is emitted.<br>
+ * <b>Properties</b><br>
+ * <b>pojoClass</b>:POJO class <br>
+ * <b>(optional)fieldMappingString</b>String of format
+ * fieldNameInJson:fieldNameInPOJO:DataType<br>
+ * <b>Ports</b> <br>
+ * <b>in</b>:input tuple as a String. Each tuple represents a json string<br>
+ * <b>out</b>:tuples that are validated as per the user defined POJO are emitted
+ * as POJO on this port<br>
+ * <b>err</b>:tuples that could not be parsed are emitted on this port as
+ * KeyValPair<String,String><br>
+ * Key being the tuple and Val being the reason
+ * 
+ * @displayName SimpleStreamingJsonParser
+ * @category Parsers
+ * @tags json pojo parser streaming
+ */
+@InterfaceStability.Evolving
+public class StreamingJsonParser extends Parser<byte[], KeyValPair<String, String>>
+{
+  private String jsonSchema;
+  private transient JSONParser jsonParser;
+  private String fieldMappingString;
+  private List<FieldInfo> fieldInfos;
+  private List<ActiveFieldInfo> columnFieldSetters;
+  protected JsonKeyFinder finder;
+  private static final String FIELD_SEPARATOR = ":";
+  private static final String RECORD_SEPARATOR = ",";
+  private ArrayList<String> columnFields;
+  private transient Class<?> pojoClass;
+
+  /**
+   * @return POJO class
+   */
+  private Class<?> getPojoClass()
+  {
+    return pojoClass;
+  }
+
+  /**
+   * Sets the POJO class
+   */
+  public void setPojoClass(Class<?> pojoClass)
+  {
+    this.pojoClass = pojoClass;
+  }
+
+  /**
+   * Returns a string representing mapping from generic record to POJO fields
+   */
+  public String getFieldMappingString()
+  {
+    return fieldMappingString;
+  }
+
+  /**
+   * Comma separated list mapping a field in JSON schema to POJO field eg :
+   * fieldName:INTEGER,fieldName:DOUBLE
+   */
+  public void setFieldMappingString(String genericRecordToPOJOFieldsMapping)
```

Comment:
Recommended to have argument name same as field member. Make genericRecordToPOJOFieldsMapping to fieldMappingString